### PR TITLE
WEB-671: Mobile Side Nav Changes + Nav Accessibility

### PR DESCRIPTION
**Pull request by Finholt**

### CODE REVIEW

File: assets/images/dwolla-developers-logo.svg [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+  </g>
+</svg>
```
Comment (Finholt) [inline assets/images/dwolla-developers-logo.svg:16]: Switched to SVG for better image scaling

File: components/base/Button.tsx [hunk 2]
```diff
@@ -56,12 +61,20 @@ const ButtonStyle = styled.button`
...
+    text-align: center;
+  }
 
   &.primary {
     background-color: ${PURPLE_PRIMARY_BUTTON};
     border-color: transparent;
 
-    :hover {
+    :hover,
+    :focus {
```
Comment (Finholt) [inline components/base/Button.tsx:77]: Not having a focus state reduced keyboard accessibility. Applying the hover state isn't the ideal solution, but it was a quick fix. Should probably be re-addressed later

File: components/layout/SideNav.tsx [hunk 3]
```diff
@@ -241,7 +262,12 @@ const DocGroup = ({ title, docs }: DocGroupProps) => {
 
   return (
     <>
-      <GroupToggle onClick={() => setIsExpanded(!isExpanded)}>
+      <GroupToggle
+        tabIndex={0}
+        onClick={() => setIsExpanded(!isExpanded)}
+        onKeyPress={(e) =>
+          e.key === "Enter" ? setIsExpanded(!isExpanded) : false}
+      >
```
Comment (Finholt) [inline components/layout/SideNav.tsx]: Made this focusable so the links sorted under them could be accessed via keyboard

File: components/layout/SideNav.tsx [hunk 4]
```diff
@@ -316,36 +419,44 @@ const SideNav = ({ sectionLinks, pages }: SideNavProps) => {
...
-              isActive={false}
-            />
-
-            {activeSection && (
-              <SectionLink
-                linkProps={{ ...activeSection, isSection: false }}
-                isActive
-              />
+          <StickySectionWrap>
+            {activeSection && activeSection.isSection && (
```
Comment (Finholt) [inline components/layout/SideNav.tsx:442]: I found you were actually able to tab over to the links in the hidden pane, so I'm just not rendering the links themselves if the pane isn't active

File: components/base/Button.tsx [hunk 5]
```diff
@@ -17,25 +19,28 @@ import { POPPINS } from "../typography";
...
+  link?: { href: string; external: boolean };
 };
 
-function Button({ text, size, variant, type = "button" }: Props) {
-  return (
-    <ButtonStyle type={type} className={`${variant} ${size}`}>
+const Button = ({ text, size, variant, type = "button", link }: Props) =>
+  link && link.href ? (
+    <ButtonLink text={text} size={size} variant={variant} link={link} />
+  ) : (
```
Comment (Finholt) [inline components/base/Button.tsx:31]: Added a `link` prop to more easily turn these buttons into links

**Review by Finholt — COMMENTED** [review]
In hindsight, I probably should've broken the keyboard accessibility of our nav elements into its own ticket to keep this PR from getting as messy as it has. Let me know if any questions pop up as a result!

**Review by sausman — APPROVED** [review]

**Pull request merged**